policyengine_us/variables/gov/states/hi/tax/income/credits/cdcc/dependent_care_benefit/hi_cdcc_earned_income.py

```diff
@@ -0,0 +1,43 @@
+from policyengine_us.model_api import *
+
+
+class hi_cdcc_earned_income(Variable):
+    value_type = float
+    entity = Person
+    label = "Hawaii CDCC earned income"
+    defined_for = StateCode.HI
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://files.hawaii.gov/tax/forms/2022/n11ins.pdf#page=28"
+        "https://files.hawaii.gov/tax/forms/2022/n11ins.pdf#page=29"
+        "https://files.hawaii.gov/tax/legal/hrs/hrs_235.pdf#page=41"
+        "https://files.hawaii.gov/tax/forms/2022/schx_i.pdf#page=2"
+    )
+
+    def formula(person, period, parameters):
```

Comment:
I'd suggest restructuring this as follows:
* `hi_cdcc_eligible_income_floor`: Remove earnings element and add the disabled/student logic. This way it's just the floor that would apply.
* `hi_cdcc_floored_earned_income_indv`: Earnings with floor applied.

Have to run but I think something like this to start would be clearer